cmd/attest_test.go

```diff
@@ -0,0 +1,199 @@
+	ExternalTPM = rwc
+
+	//without nonce
+	RootCmd.SetArgs([]string{"attest","--key","AK"})
+	if err := RootCmd.Execute(); err == nil {
+		t.Error(err)
+	}
+
+	//with odd length nonce
+	RootCmd.SetArgs([]string{"attest","--nonce","12345","--key","AK"})
+	if err := RootCmd.Execute(); err == nil {
+		t.Error(err)
+	}
+
+}
+
+func TestAttestPass(t *testing.T) {
+	rwc := test.GetTPM(t)
+	defer client.CheckedClose(t, rwc)
+	ExternalTPM = rwc
+
+	tests := []struct {
+		name          string
+		key           string
+		algo          string
+		nonce         string
+		teenonce      string
+		outputText    string
+	}{
+		{"DefaultKey","","rsa","2222","",""},
+		{"Withnonce", "AK", "rsa", "2222", "", ""},
+		{"WithTextOutput", "AK", "rsa", "2222", "", "text1111"},
+		//@TODO These cases works when we are able to create gce key with ExternalTPM.
+		// {"WithNonce","gceAK","rsa","2222","",""},
+		// {"WithTextOutput","gceAK","rsa","2222","","text1111"},
+	}
+
+	for _, op := range tests {
+		t.Run(op.name, func(t *testing.T) {
+			secretFile1 := makeOutputFile(t, "attest.binarypb")
+
+			attestArgs := []string{"attest", "--output", secretFile1}
+			if op.key != "" {
+				attestArgs = append(attestArgs, "--key", op.key)
+			}
+			if op.algo != "" {
+				attestArgs = append(attestArgs, "--algo", op.algo)
+			}
+			if op.nonce != "" {
+				attestArgs = append(attestArgs, "--nonce", op.nonce)
+			}
+			if op.teenonce != "" {
+				attestArgs = append(attestArgs, "--teenonce", op.teenonce)
+			}
+			if op.outputText != "" {
+				attestArgs = append(attestArgs, "--outputText", op.outputText)
+			}
+
+			RootCmd.SetArgs(attestArgs)
+			if err := RootCmd.Execute(); err != nil {
+				t.Error(err)
+			}
+
+		})
+	}
+}
+
+func TestGceFlagFail(t *testing.T) {
+	rwc := test.GetTPM(t)
+	defer client.CheckedClose(t, rwc)
+	ExternalTPM = rwc
+	//@TODO This case works when we are able to create gce key with ExternalTPM.
+ 
+	// RootCmd.SetArgs([]string{"attest", "--nonce", "1111","--key","gceAK", "--gce-zone", "us-west1-b", "--gce-project-id",
+	// 	"sev-team-project"})
+	// if err := RootCmd.Execute(); err != nil {
+	// 	t.Error(err)
+	// }
+}
+
+func TestGCEFlags(t *testing.T) {
+	rwc := test.GetTPM(t)
+	defer client.CheckedClose(t, rwc)
+	ExternalTPM = rwc
+
+	tests := []struct {
+		name             string
+		nonce            string
+		key              string
+		gceProjectId     string
+		gceProjectNumber string
+		gceZone          string
+		gceInstanceId    string
+		gceInstanceName  string
+	}{
+		//@TODO These cases works when we are able to create gce key with ExternalTPM.
+
+		// {"ProjectMetadataWithAK","1111","gceAK","sev-team-project","1071581149866","","",""},
+		// {"ProjectMetadataWithAK","1111","gceAK","sev-team-project","1071581149866","us-west1-b","",""},
+	}
+
+	for _, op := range tests {
+		t.Run(op.name, func(t *testing.T) {
+
+			attestFlagsArgs := []string{"attest", "--nonce", op.nonce, "--key", op.key}
+			if op.gceProjectId != "" {
```

Comment:
I want to understand how to use the reflect package appropriately so that I can use the learnings for go-tdx-guest work as well.  I will raise another request once reflect changes are done. I hope that works since they are more from a readability perspective and functionality will remain the same.